packages/cypress/src/migrations/update-17-2-0/add-nx-metadata.ts

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import {
+  createProjectGraphAsync,
+  formatFiles,
+  getProjects,
+  glob,
+  readProjectConfiguration,
+  TargetConfiguration,
+  Tree,
+} from '@nx/devkit';
+import { createNodes, NxCypressMetadata } from '../../plugins/plugin';
+import {
+  createProjectRootMappingsFromProjectConfigurations,
+  findProjectForPath,
+} from 'nx/src/project-graph/utils/find-project-for-path';
+import { CypressExecutorOptions } from '../../executors/cypress/cypress.impl';
+import { dirname } from 'path';
+
+export default async function update(tree: Tree) {
+  const configFiles = glob(tree, [createNodes[0]]);
+
+  const proj = Object.fromEntries(getProjects(tree).entries());
+
+  const rootMappings = createProjectRootMappingsFromProjectConfigurations(proj);
+
+  for (const configFile of configFiles) {
+    const siblings = tree.children(dirname(configFile));
+    if (!siblings.includes('project.json')) {
+      continue;
+    }
+
+    const projectName = findProjectForPath(configFile, rootMappings);
+    const projectConfig = readProjectConfiguration(tree, projectName);
+    const e2eTarget: TargetConfiguration<CypressExecutorOptions> =
+      projectConfig.targets?.e2e;
+
+    if (!e2eTarget || e2eTarget.executor !== '@nx/cypress:cypress') {
+      continue;
+    }
+
+    const nxMetadata: NxCypressMetadata = {};
+
+    nxMetadata.devServerTarget = e2eTarget.options?.devServerTarget;
```

Comment:
I'm wondering if having these three versions of `devServerTarget` makes sense or not. It's fixed to what we generate for users, but they customize the setup.

Maybe something like:

```js
export const nx: NxCypressMetadata = {
  devServerTarget: process.env.NX_CYPRESS_DEV_SERVER_TARGET ?? 'myapp:serve'
}
```

We can also skip the whole `nx` metadata, and document that users can change the target with `NX_CYPRESS_DEV_SERVER_TARGET` or something. Similar thing for ports, I think expecting `PORT` to be set is pretty standard

Or something like this if we are keen to generate separate `configuration`.

```js
export const nx: NxCypressMetadata = {
  devServerTarget: 'myapp:serve',
  configurations: {
    production: 'myapp:serve:production',
    ci: 'myapp:static-serve',
  }
}
```

So it's more explicit about generating support for `--configuration=production` and `--configuration=ci`, and is able to support additional executor options if we want to.